controllers/auth-controller.js

```diff
@@ -34,25 +35,69 @@ const register = async (req, res) => {
   }
 
   const hashPassword = await bcrypt.hash(password, 10);
+  const verificationToken = nanoid();
+  console.log(verificationToken);
 
   const newUser = await User.create({
     ...req.body,
     avatarURL,
     password: hashPassword,
+    verificationToken,
   });
 
+  const verifyEmail = createVerifyEmail(email, verificationToken);
+
+  await sendEmail(verifyEmail);
+
   res.status(201).json({
     email: newUser.email,
     subscription: newUser.subscription,
   });
 };
 
+const verifyEmail = async (req, res) => {
+  const { verificationToken } = req.params;
+  const user = await User.findOne({ verificationToken });
+  if (!user) {
+    throw HttpError(404, "User not found");
+  }
+  await User.findByIdAndUpdate(user._id, {
+    verify: true,
+    verificationToken: "",
```

Comment:
token should be null

Reply:
if it returns error, delete `userSchema.pre("findOneAndUpdate", validateAtUpdate);` this code in user model

Reply:
I fixed the mistake